libs/MeshKernel/tests/src/CurvilinearGridGenerationTests.cpp

```diff
@@ -0,0 +1,358 @@
+{
+
+    mk::CurvilinearGridGenerateCircularGrid generate;
+
+    mk::MakeGridParameters parameters = {
+        .num_columns = 10,
+        .num_rows = 12,
+        .angle = 0.0,
+        .block_size_x = 10.0,
+        .block_size_y = 20.0,
+        .uniform_columns_fraction = 1.0,
+        .uniform_rows_fraction = 0.0};
+
+    auto grid = generate.Compute(parameters, mk::Projection::cartesian);
+
+    std::vector<double> gradedYValues{0.0,
+                                      31.6763921753316,
+                                      71.5410984881053,
+                                      121.710789550376,
+                                      184.849293106268,
+                                      264.309033576453,
+                                      364.309033576453,
+                                      490.158928640636,
+                                      648.540889517293,
+                                      847.864421081162,
+                                      1098.71287639251,
+                                      1414.40539417197,
+                                      1811.7040965229};
+
+    ASSERT_EQ(grid.NumM(), static_cast<mk::UInt>(parameters.num_columns + 1));
+    ASSERT_EQ(grid.NumN(), static_cast<mk::UInt>(parameters.num_rows + 1));
+
+    const double tolerance = 1.0e-10;
+
+    for (mk::UInt i = 0; i < grid.NumN(); ++i)
+    {
+        double x = 0.0;
+
+        for (mk::UInt j = 0; j < grid.NumM(); ++j)
+        {
+            EXPECT_NEAR(x, grid.GetNode(i, j).x, tolerance);
+            EXPECT_NEAR(gradedYValues[i], grid.GetNode(i, j).y, tolerance);
+            x += parameters.block_size_x;
+        }
+    }
+}
+
+TEST(CurvilinearGridGenerationTests, GradedXUniformYGrid)
+{
+
+    mk::CurvilinearGridGenerateCircularGrid generate;
+
+    mk::MakeGridParameters parameters = {
+        .num_columns = 10,
+        .num_rows = 12,
+        .angle = 0.0,
+        .block_size_x = 10.0,
+        .block_size_y = 20.0,
+        .uniform_columns_fraction = 0.0,
+        .uniform_rows_fraction = 1.0};
+
+    auto grid = generate.Compute(parameters, mk::Projection::cartesian);
+
+    std::vector<double> gradedXValues{0.0,
+                                      38.2362245665865,
+                                      67.4764019487152,
+                                      89.8370817237131,
+                                      106.93684119048,
+                                      120.013446050598,
+                                      133.090050910717,
+                                      150.189810377484,
+                                      172.550490152481,
+                                      201.79066753461,
+                                      240.026892101197};
+
+    ASSERT_EQ(grid.NumM(), static_cast<mk::UInt>(parameters.num_columns + 1));
+    ASSERT_EQ(grid.NumN(), static_cast<mk::UInt>(parameters.num_rows + 1));
+
+    const double tolerance = 1.0e-10;
```

Comment:
constexpr

Reply:
Will change, but change is not necessary. The value is needed only at runtime.